cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/worker/TaskWorkerServiceLauncher.java

```diff
@@ -269,6 +270,21 @@ public void run() {
             }
           }
 
+          if (cConf.getBoolean(Constants.TaskWorker.TASK_LIVENESS_PROBE_ENABLED)) {
+            if (twillPreparer instanceof ExtendedTwillPreparer) {
```

Comment:
Should the check be for instanceof `KubeTwillPreparer`?

Reply:
ideally yes, but It's not a part of master-spi.
`KubeTwillPreparer` is only used in cdap-kubernetes repo .